File: 백준/Silver/2167. 2차원 배열의 합/2차원 배열의 합.java
import java.util.Scanner;

class Main{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int col = sc.nextInt();
        int row = sc.nextInt();
        int [][] map = new int[col][row];
        for(int i = 0; i < col; i++){
            for(int j = 0; j < row; j++){
                map[i][j] = sc.nextInt();
            }
        }
        int count, startX, startY, distX, distY;
        count = sc.nextInt();
        for(int i = 0 ; i < count ; i++){
            int result = 0;
            startX= sc.nextInt();
            startY = sc.nextInt();
            distX = sc.nextInt();
            distY = sc.nextInt();
            for(int j = startX-1; j < distX; j++){
                for(int k = startY-1; k < distY; k++){
                    result += map[j][k];
                }
            }
            System.out.println(result);
        }
        sc.close();
    }
}

File: 백준/Silver/2167. 2차원 배열의 합/2차원 배열의 합.cc
#include <iostream>
#include <queue>
#include <vector>
using namespace std;

int main(){
    int col, row;
    cin >> col >> row;
    int** map = new int*[col];
    for(int i = 0 ; i < col; i++) {
        map[i] = new int[row];
        for(int j = 0; j < row; j++){
            cin >> map[i][j];
        }
    }
    int count, startX,startY,distX,distY;
    cin >> count;
    for(int i = 0 ; i < count ;i++){
        int result = 0;
        cin >> startX >> startY >> distX >> distY;
        for(int j = startX-1; j < distX; j++){
            for(int k = startY-1; k < distY; k++){
                result += map[j][k];
            }
        }
        cout << result << endl;
    }
}